rs/ic_os/config/src/types.rs

```diff
@@ -0,0 +1,76 @@
+use ic_types::malicious_behaviour::MaliciousBehaviour;
+use serde::{Deserialize, Serialize};
+use std::net::{Ipv4Addr, Ipv6Addr};
+use std::path::PathBuf;
+use url::Url;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct SetupOSConfig {
+    pub network_settings: NetworkSettings,
+    pub icos_settings: ICOSSettings,
+    pub hostos_settings: HostOSSettings,
+    pub guestos_settings: GuestOSSettings,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct HostOSConfig {
+    pub network_settings: NetworkSettings,
+    pub icos_settings: ICOSSettings,
+    pub hostos_settings: HostOSSettings,
+    pub guestos_settings: GuestOSSettings,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct GuestOSConfig {
+    pub network_settings: NetworkSettings,
+    pub icos_settings: ICOSSettings,
+    pub guestos_settings: GuestOSSettings,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct HostOSSettings {
+    pub vm_memory: u32,
+    pub vm_cpu: String,
+    pub verbose: bool,
+}
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NetworkSettings {
+    pub ipv6_prefix: Option<Ipv6Addr>,
+    pub ipv6_address: Option<Ipv6Addr>,
+    pub ipv6_gateway: Ipv6Addr,
+    pub ipv4_address: Option<Ipv4Addr>,
+    pub ipv4_gateway: Option<Ipv4Addr>,
+    pub ipv4_prefix_length: Option<u8>,
+    pub domain: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ICOSSettings {
+    pub nns_public_key_path: PathBuf,
+    pub nns_urls: Vec<Url>,
+    pub elasticsearch_hosts: String,
+    // help: elasticsearch_tags is a dev field?
+    pub elasticsearch_tags: Option<String>,
```

Comment:
Is elasticsearch_tags a dev field?

Reply:
Now is a good opportunity to make these elasticsearch things into its own Logging structure.

Reply:
We will look into type system refining in a follow-up PR. We are going to do a round of refining before rolling out to HostOS and GuestOS 👍